### ISSUE TITLE
Undefined array property access in if statement not detected

### ISSUE THREAD

**Issue by piehei** [facebook/flow#5028]
Hi

I'm somewhat new to Flow. This could be a bug or I'm missing something. Please educate me!

If you look at this code ([Try Flow](https://flow.org/try/#0MYewdgzgLgBAhgJwTAXDAgkuBPAPGAVwFsAjAUwQD4YBeGAbQEYAaAJmYGYBdAbgCg+ASwBmMABSIEAOgA2ZMAHMoAC1o06jVgEoYAbz4wYoSCDmyQCsQHIptq1v4BfASPGTZ8gB5qN2vQaNwCFMyc0sbOwc+Zz45WDhaeCQPME8eIA)):
```

const arr : Array<number> = [1,2,3];

if (arr.length === 12) {
  console.log('...');
}

if (arr.lenx === 12) {
  console.log('...');
}

let a = arr.lenx;
```

The only error Flow finds is from the last line. Property lenx not found in arr which is of type Array. My gut says Flow should see the problem in the second comparison but it doesn't. 

Why is Flow not detecting the if (arr.lenx === 12) erroneous property access? Should it, and this is a bug, or is this by design? 

I was bitten by a bug where I misspelled arr.lenght instead of arr.length and was surprised to realise that Flow didn't warn me. Thanks a lot in advance!

**Comment by CharlesStover** [comment]
This is likely by design. You should be able to check for things, such as:

```
if (myArray.myPrototype === undefined) { }
```

And you can create a custom method like so:

```
Array.prototype.lenx = function() { return this.length; }
console.log([1, 2, 3].lenx());
```

**Comment by popham** [comment]
It's by design. See [typing/statement.ml#L4301-L4304](https://github.com/facebook/flow/blob/5149c427339e9d838dc02afa7f296f9df9700520/src/typing/statement.ml#L4301-L4304).

**Comment by Mathspy** [comment]
@goodmind This now does what is expected: [TryFlow](https://flow.org/try/#0MYewdgzgLgBAhgJwTAXDAgkuBPAPGAVwFsAjAUwQD4YBeGAbQEYAaAJmYGYBdAbgCg+ASwBmMABSIEAOgA2ZMAHMoAC1o06jVgEoYAbz4wYoSCDmyQCsQHIptq1v4BfASPGTZ8gB5qN2vQaNwCFMyc0sbOwc+Zz45WDhaeCQPME8eIA)

Fixed in [v0.68.0](https://github.com/facebook/flow/releases/tag/v0.68.0)
